src/main/java/bio/terra/cli/command/resource/update/GcsObject.java

```diff
@@ -50,8 +53,10 @@ protected void execute() {
     UpdateReferencedGcsObjectParams gcsObjectParams =
         new UpdateReferencedGcsObjectParams.Builder()
             .resourceFields(updateResourceParams)
-            .bucketName(bucketResourceUpdate.getNewBucketName())
-            .objectName(newObjectName)
+            .bucketName(
+                Optional.ofNullable(newBucketName.getNewBucketName())
+                    .orElse(resource.getBucketName()))
+            .objectName(Optional.ofNullable(newObjectName).orElse(resource.getObjectName()))
```

Comment:
this is to allow user to specify only one properties on the CLI side to update the reference target even though the WSM api requires both.

Reply:
ah I see. then I think the right place to include that logic (i.e. WSM API requires both even if they are not being changed) is in `WorkspaceManagerService` when we translate the `UpdateReferencedGcsObjectParams` into the WSM API request object.

Reply:
hmm it's harder to get the original bucket name or object name in the `WorkspaceManagerService` without calling the resourceApi to get the original bucket object though. do you suggest passing in the original resource  `resource.updateReferenced(gcsObjectParams, resource);`

Reply:
ah good point. How about we add two new fields to the update params class that can hold these properties always: `originalBucketName` and `originalObjectName`? I think that's a little clearer, because otherwise this is inconsistent with all other update classes where `null` means "no change". A comment explaining that this is a limitation of the wsm api would also be helpful.

side note: Why does WSM require this -- is it possible to change that in the future?

Reply:
Yeah. I didn't realize this Api causes inconvenience from the client side. We could change it on the WSM side in the future. Currently the `UpdateGcsBucketObjectReferenceRequestBody` reuses the `GcpGcsObjectAttributes` which requires both bucket name and object name to be present. `GcpGcsObjectAttributes` was originally used for creating a reference to the bucket object which makes sense to have both field to be present.